packages/kbn-search-connectors/utils/to_alphanumeric.ts

```diff
@@ -11,5 +11,5 @@ export const toAlphanumeric = (input: string) =>
   input
     .trim()
     .replace(/[^a-zA-Z0-9]+/g, '-') // Replace all special/non-alphanumerical characters with dashes
-    .replace(/^[-]+|[-]+$/g, '') // Strip all leading and trailing dashes
+    .replace(/(^-+|(?<!-)-+$)/g, '') // Strip all leading and trailing dashes
```

Comment:
Testing this with https://regexr.com/ I see an error that negative lookbehind may not be supported in all browsers.

Reply:
Damn, it was the officially recommended workaround:

> Avoid this problem by rewriting the regular expression to not contain the ambiguity about when to start matching whitespace sequences. For instance, by using a negative look-behind (/^\s+|(?<!\s)\s+$/g), or just by using the built-in trim method (text.trim()).

Unfortunately there's no `.trim()` for `-` characters.

Reply:
https://caniuse.com/js-regexp-lookbehind

Reply:
might be ok we should just cross reference support with out supported browsers.